src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -25,27 +49,58 @@ public void periodic() {
 
     @Override
     public void simulationPeriodic() {
-        Translation2d goalDiff = FieldMap.goalCenterPoint.minus(driveTrain.getPose().getTranslation());
-        double distance = goalDiff.getNorm();
-        targetInfoSim[3] = distance / Constants.inchToMetersConversionFactor;
+        if (getMode() == VisionMode.GOALFINDER) {
+            Translation2d goalDiff = FieldMap.goalCenterPoint.minus(driveTrain.getPose().getTranslation());
+            double distance = goalDiff.getNorm() / Constants.inchToMetersConversionFactor;
+
+            double angleRobotGoal = Math.atan2(goalDiff.getY(), goalDiff.getX());
+            double visionAngle = Math.toRadians(driveTrain.getHeading()) - angleRobotGoal;
+
+            if ( distance < MAX_DISTANCE && Math.abs(visionAngle) < HORIZONTAL_HALF_FOV) {
+                // target should be visible to the robot
+                targetInfoSim[1] = 1;
+                targetInfoSim[3] = distance;
+                targetInfoSim[4] = visionAngle;
+            } else {
+                targetInfoSim[1] = 0;
+                targetInfoSim[3] = 0;
+                targetInfoSim[4] = 0;
+            }
+        } else {
+            targetInfoSim[1] = 1;
+            targetInfoSim[3] = 0;
+            targetInfoSim[4] = 0;
+        }
+
+        // include the heartbeat signal that Vision puts in
+        targetInfoSim[0] = System.nanoTime() * 1e-9;
```

Comment:
This should use new Robot.time() method.

Reply:
Fixed now